src/utils.rs

```diff
@@ -292,93 +292,85 @@ mod tests {
     m.iter().map(|x| to_F_vec(x.clone())).collect()
   }
 
-  #[test]
-  fn test_vector_add() {
-    let a = to_F_vec::<Fq>(vec![1, 2, 3]);
-    let b = to_F_vec::<Fq>(vec![4, 5, 6]);
-    let res = vector_add(&a, &b);
-    assert_eq!(res, to_F_vec::<Fq>(vec![5, 7, 9]));
+  fn test_vector_add_with<F: PrimeField>() {
+    let a = to_F_vec::<F>(vec![1, 2, 3]);
+    let b = to_F_vec::<F>(vec![4, 5, 6]);
+    let res = vector_add::<F>(&a, &b);
+    assert_eq!(res, to_F_vec::<F>(vec![5, 7, 9]));
   }
 
-  #[test]
-  fn test_vector_elem_product() {
-    let a = to_F_vec::<Fq>(vec![1, 2, 3]);
-    let e = Fq::from(2);
+  fn test_vector_elem_product_with<F: PrimeField>() {
+    let a = to_F_vec::<F>(vec![1, 2, 3]);
+    let e = F::from(2);
     let res = vector_elem_product(&a, &e);
-    assert_eq!(res, to_F_vec::<Fq>(vec![2, 4, 6]));
+    assert_eq!(res, to_F_vec::<F>(vec![2, 4, 6]));
   }
 
-  #[test]
-  fn test_matrix_vector_product() {
+  fn test_matrix_vector_product_with<F: PrimeField>() {
     let matrix = vec![vec![1, 2, 3], vec![4, 5, 6]];
     let vector = vec![1, 2, 3];
-    let A = to_F_matrix::<Fq>(matrix);
-    let z = to_F_vec::<Fq>(vector);
+    let A = to_F_matrix::<F>(matrix);
+    let z = to_F_vec::<F>(vector);
     let res = matrix_vector_product(&A, &z);
 
-    assert_eq!(res, to_F_vec::<Fq>(vec![14, 32]));
+    assert_eq!(res, to_F_vec::<F>(vec![14, 32]));
   }
 
-  #[test]
-  fn test_hadamard_product() {
-    let a = to_F_vec::<Fq>(vec![1, 2, 3]);
-    let b = to_F_vec::<Fq>(vec![4, 5, 6]);
+  fn test_hadamard_product_with<F: PrimeField>() {
+    let a = to_F_vec::<F>(vec![1, 2, 3]);
+    let b = to_F_vec::<F>(vec![4, 5, 6]);
     let res = hadamard_product(&a, &b);
-    assert_eq!(res, to_F_vec::<Fq>(vec![4, 10, 18]));
+    assert_eq!(res, to_F_vec::<F>(vec![4, 10, 18]));
   }
 
-  #[test]
-  fn test_matrix_vector_product_sparse() {
+  fn test_matrix_vector_product_sparse_with<G: Group>() {
     let matrix = vec![
-      (0, 0, Fq::from(1)),
-      (0, 1, Fq::from(2)),
-      (0, 2, Fq::from(3)),
-      (1, 0, Fq::from(4)),
-      (1, 1, Fq::from(5)),
-      (1, 2, Fq::from(6)),
+      (0, 0, G::Scalar::from(1u64)),
+      (0, 1, G::Scalar::from(2u64)),
+      (0, 2, G::Scalar::from(3u64)),
+      (1, 0, G::Scalar::from(4u64)),
+      (1, 1, G::Scalar::from(5u64)),
+      (1, 2, G::Scalar::from(6u64)),
     ];
 
-    let z = to_F_vec::<Fq>(vec![1, 2, 3]);
-    let res =
-      matrix_vector_product_sparse::<Ep>(&SparseMatrix::<Ep>::with_coeffs(2, 3, matrix), &z);
+    let z = to_F_vec::<G::Scalar>(vec![1, 2, 3]);
+    let res = matrix_vector_product_sparse::<G>(&SparseMatrix::<G>::with_coeffs(2, 3, matrix), &z);
```

Comment:
This is not uniquely related to this PR but also to previous commits I guess, but since is my first time reviewing this file I will comment it here.

`SparseMatrix` does not need `G` but just `G::Scalar`, thus maybe makes more sense to abstract it around `F` instead of aroung `G`.

This also applies to the rest of methods of `SparseMatrix` and to the standalone methods `sparse_vec_to_mle` and `matrix_vector_product_sparse`.
In the same way that other methods of `src/utils.rs` are with generic `F`, such as `dense_vec_to_mle`, `matrix_vector_product`, etc.

I mention this in this PR because this would then affect also the tests modified in this PR, where instead of for example
`test_matrix_vector_product_sparse_with<G: Group>()`
it could be
`test_matrix_vector_product_sparse_with<F: PrimeField>()`

Reply:
Update: added this description into a issue (#32 ) so we can track this.

Reply:
Update2: it has been addressed at https://github.com/privacy-scaling-explorations/Nova/pull/33

Reply:
Cherry picked commit into this branch